solidity/contracts/token/ERC20WithPermit.sol

```diff
@@ -0,0 +1,221 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity <0.9.0;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+import "./IERC20WithPermit.sol";
+
+/// @title  ERC20WithPermit
+/// @notice Burnable ERC20 token with EIP2612 permit functionality. User can
+///         authorize a transfer of their token with a signature conforming
+///         EIP712 standard instead of an on-chain transaction from their
+///         address. Anyone can submit this signature on the user's behalf by
+///         calling the permit function, as specified in EIP2612 standard,
+///         paying gas fees, and possibly performing other actions in the same
+///         transaction.
+/// @dev    By the time of implementing this contract there is no other audited
+///         EIP2612 token implementation other than Uniswap V2 on which this
+///         code is based on.
+contract ERC20WithPermit is IERC20WithPermit, Ownable {
+    using SafeMath for uint256;
+
+    uint8 public constant decimals = 18;
+
+    string public name;
+    string public symbol;
+
+    uint256 public override totalSupply;
+    mapping(address => uint256) public override balanceOf;
+    mapping(address => mapping(address => uint256)) public override allowance;
+
+    uint256 public cachedChainId;
+    bytes32 public cachedDomainSeparator;
+
+    // keccak256("Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)");
```

Comment:
I see this comment in the original uniswap implementation and the covpools implementation. What does it mean?